internal/tasks/manifests_test.go

```diff
@@ -724,7 +724,21 @@ func TestCheckVulnerabilitiesForNextManifestWithError(t *testing.T) {
 	expectSuccess(t, ExecuteOne(j.CheckVulnerabilitiesForNextManifest()))
 	expectError(t, sql.ErrNoRows.Error(), ExecuteOne(j.CheckVulnerabilitiesForNextManifest()))
 	tr.DBChanges().AssertEqualf(`
-		UPDATE vuln_info SET status = 'Low', next_check_at = %[2]d, checked_at = %[3]d, index_finished_at = %[3]d WHERE repo_id = 1 AND digest = '%[1]s';
-	`, image.Manifest.Digest, s.Clock.Now().Add(60*time.Minute).Unix(), s.Clock.Now().Unix())
+		UPDATE vuln_info SET status = '%[4]s', next_check_at = %[2]d, checked_at = %[3]d, index_finished_at = %[3]d WHERE repo_id = 1 AND digest = '%[1]s';
+	`, image.Manifest.Digest, s.Clock.Now().Add(60*time.Minute).Unix(), s.Clock.Now().Unix(), clair.LowSeverity)
 	assert.DeepEqual(t, "delete counter", claird.IndexDeleteCounter, 1)
+
+	// also the clair configuration was updated to make transient errors less likely to happen
+	claird.IndexState = "a8b9e94aa9c8e4bb2818af1f52507b0b"
+	expectSuccess(t, j.CheckClairManifestState())
+	tr.DBChanges().AssertEqualf(`
+		UPDATE vuln_info SET status = '%[3]s', next_check_at = %[2]d WHERE repo_id = 1 AND digest = '%[1]s';
+	`, image.Manifest.Digest, s.Clock.Now().Unix(), clair.PendingVulnerabilityStatus)
+
+	s.Clock.StepBy(1 * time.Minute)
+	expectSuccess(t, ExecuteOne(j.CheckVulnerabilitiesForNextManifest()))
+	expectError(t, sql.ErrNoRows.Error(), ExecuteOne(j.CheckVulnerabilitiesForNextManifest()))
+	tr.DBChanges().AssertEqualf(`
+		UPDATE vuln_info SET status = '%[4]s', next_check_at = %[2]d, checked_at = %[3]d WHERE repo_id = 1 AND digest = '%[1]s';
```

Comment:
The `index_state` column for this `vuln_info` is filled in line 704, but then changed in the Clair mock in line 732. Should it not be updated to match the new value here?

Reply:
Good catch! One query was missing updating a value.